tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/JmapSettingsSetMethodContract.scala

```
@@ -661,4 +661,259 @@ trait JmapSettingsSetMethodContract {
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract()
+      .jsonPath()
+
+    val afterSettingsSetState: UuidState = server.getProbe(classOf[JmapSettingsProbe])
+      .getLatestState(BOB)
+
+    assertThat(response.getString("methodResponses[0][1].oldState")).isEqualTo(JmapSettingsStateFactory.INITIAL.serialize)
+    assertThat(response.getString("methodResponses[0][1].oldState")).isNotEqualTo("methodResponses[0][1].newState")
+    assertThat(response.getString("methodResponses[0][1].newState")).isEqualTo(afterSettingsSetState.serialize)
+  }
+
+  @Test
+  def updateShouldFailWhenEmptyBothUpdatePartialAndFullResetPatch(): Unit =
+      `given`
+      .body(
+        s"""{
+           |	"using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:settings"],
+           |	"methodCalls": [
+           |		[
+           |			"Settings/set",
+           |			{
+           |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |				"update": {
+           |					"singleton": {}
+           |				}
+           |			}, "c1"
+           |		]
+           |	]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .body("methodResponses[0]", jsonEquals(
+        s"""[
+           |    "Settings/set",
+           |    {
+           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |        "oldState": "${JmapSettingsStateFactory.INITIAL.serialize}",
+           |        "newState": "${JmapSettingsStateFactory.INITIAL.serialize}",
+           |        "notUpdated": {
+           |            "singleton": {
+           |                "type": "invalidArguments",
+           |                "description": "update request must not be empty"
+           |            }
+           |        }
+           |    },
+           |    "c1"
+           |]""".stripMargin))
+
```

TODO
update partial should fail when invalid request (ex: not a JsString)